crates/web5/src/dids/methods/did_web/mod.rs

```diff
@@ -18,6 +21,29 @@ pub struct DidWeb {
 }
 
 impl DidWeb {
+    pub fn new(domain: &str, public_jwk: Jwk) -> Result<Self> {
+        let did = format!("did:web:{}", domain);
+
+        let verification_method = VerificationMethod {
+            id: format!("{}#key-0", did),
```

Comment:
I'm slightly troubled by the hard coding of `#key-0` here, but actually moreso with the assumption that all did:web's shall be created initially with a verification method, because the [did:web spec](https://w3c-ccg.github.io/did-method-web/) doesn't assert requirements for the existence of a verification method. This is in contrast with did:dht and did:jwk which both assert requirements of the verification method at time of creation. 

However, we do have the requirement of:

> There must be at least one Verification Method in each DID Document.

(from [our web5-spec](https://github.com/TBD54566975/web5-spec/blob/main/spec/did.md#did-document-data-model)).

So in this case, I think it's right to embrace our spec, because I support the idea that creation of a `did:web` without at least one verification method is non-sensical.

I do wonder if instead we should use a JWK thumbprint for the fragment, but I can see you're using the examples from the did:web spec as inspiration here, which use `#key-0`.

So alright, I've talked myself into this being the ideal way forward. I'm going to update to APID to reflect the constructor here.

Reply:
@diehuxx I have updated the APID, reminder to pull latest.

Reply:
I am of two minds here:
1. I think DIDs without verification methods are silly
2. The spec allows for them and so should we

So, we have a few options on how we can handle this
1. Be prescriptive with create but lenient with read/our data model 

This means we can consume DID webs in the wild that may not have VMs, but our impl will only facilitate the creation of DID webs with VMs


2. Allow DID webs to be created / read according to the did:web spec (no VMs needed)

3. Mandate a VM is used to create a DID Web

Here we run into the comments you've raised @KendallWeihe -- what should the first KID be? `#key-0` is a conventional and reasonable choice, and we could expose an option for a user to override it


----------------------------------

zooming out again this really depends on how we're conceptualizing the rust SDK -- is it TBD's opinionated usage of standards? or is it an open source project that provides relatively unopinionated/plain implementations of standards?

Reply:
> Be prescriptive with create but lenient with read/our data model

This seems to be the optimal compromise. We offer support at the data model (and we have an [open PR](https://github.com/TBD54566975/web5-rs/pull/233) on this, which at the moment is serving as a reminder to get to when we have the bandwidth), and then our functions offer an opinionated view. I suppose we can offer an override for the `#key-0` value, but I've been hesitant to offer any optional parameters this until we are further along in our implementation for the sake of agility.

Reply:
Sounds like the consensus is to leave `#key-0` as the default for new and add the ability to add a custom value in the future. Here's a ticket tracking this: https://github.com/TBD54566975/web5-rs/issues/267